src/main/java/tech/toparvion/analog/service/choice/ChoicesAutoReloadConfiguration.java

```diff
@@ -0,0 +1,96 @@
+package tech.toparvion.analog.service.choice;
+
+import io.micrometer.core.instrument.util.NamedThreadFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import tech.toparvion.analog.model.config.ChoicesAutoReloadProperties;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.WatchService;
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static java.nio.file.Files.exists;
+import static java.nio.file.Files.isDirectory;
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static java.util.Objects.requireNonNullElse;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+
+/**
+ * Configuration for choices list auto reloading
+ *
+ * @author Polyudov
+ * @since 0.14
+ */
+@Configuration
+@ConditionalOnChoicesAutoReloadEnabled
+class ChoicesAutoReloadConfiguration {
+  private static final Logger log = LoggerFactory.getLogger(ChoicesAutoReloadConfiguration.class);
+  static final String CHOICES_HOT_RELOAD_EXECUTOR = "choicesAutoReloadExecutor";
+
+  @Bean(CHOICES_HOT_RELOAD_EXECUTOR)
+  Executor forthcomingExecutor() {
+    return newSingleThreadExecutor(new NamedThreadFactory("auto-reload"));
+  }
+
+  @Bean
+  @Nullable
+  FileWatcherProvider fileWatcherProvider(ChoicesAutoReloadProperties choiceProperties) {
+    String choicesPropertiesLocation = choiceProperties.getLocation();
+    if (isNullOrEmpty(choicesPropertiesLocation)) {
+      log.info("Custom path for choices list is not present");
+      return null;
+    }
+
+    try {
+      Path choicesPropertiesPath = Paths.get(choicesPropertiesLocation); //check exception
+
+      if (!exists(choicesPropertiesPath)) {
+        log.warn("File '{}' does not exists", choicesPropertiesLocation);
+        return null;
+      }
+      if (isDirectory(choicesPropertiesPath)) {
+        log.warn("'choices.auto-reload.path' ('{}') is a directory", choicesPropertiesLocation);
+        return null;
+      }
+      return new FileWatcherProvider(choicesPropertiesPath);
+    } catch (Exception e) {
+      log.warn("Can`t create choices list watcher because of error: {}", e.getMessage());
```

Comment:
May be it's better to print all the error details since we don't know beforehand where it is coming from?
```suggestion
      log.warn("Can`t create choices list watcher", e);
```